models/classes/StatisticalMetadata/Import/Processor/ImportProcessor.php

```diff
@@ -0,0 +1,231 @@
+        $possibleMetadataToBindCount = 0;
+        $errors = 0;
+
+        // Since the metadata map has been checked for uniqueness, $metadataProperties contains only one value
+        foreach ($metadataMap as $metadata => $metadataProperties) {
+            if (empty($record[$metadata])) {
+                continue;
+            }
+
+            ++$possibleMetadataToBindCount;
+
+            // @TODO Extract to a separate method
+            try {
+                $this->metadataToBindValidator->validateRelationToResource(
+                    $resource,
+                    $metadataProperties[0][StatisticalMetadataMapper::KEY_DOMAIN]
+                );
+                $this->metadataToBindValidator->validateMetadataValue($record[$metadata]);
+            } catch (InvalidArgumentException $exception) {
+                $reports[] = Report::createWarning(
+                    sprintf(
+                        'Metadata "%s": %s',
+                        $metadata,
+                        $exception->getMessage()
+                    )
+                );
+                ++$errors;
+
+                continue;
+            }
+
+            $metadataToBind[$metadataProperties[0][StatisticalMetadataMapper::KEY_URI]] = $record[$metadata];
+        }
+
+        if ($errors) {
+            // @FIXME Throw an exception instead of error/warning
+            $this->reports[] = $possibleMetadataToBindCount === $errors
+                ? Report::createError(sprintf('Line %d: import failed', $line), null, $reports)
+                : Report::createWarning(sprintf('Line %d: partially imported', $line), null, $reports);
+        }
+
+        return $metadataToBind;
+    }
+
+    /**
+     * @throws tao_models_classes_dataBinding_GenerisInstanceDataBindingException
+     */
+    private function bindProperties(core_kernel_classes_Resource $resource, array $values): void
+    {
+        $binder = new tao_models_classes_dataBinding_GenerisInstanceDataBinder($resource);
+        $binder->bind($values);
+    }
+
+    private function buildFinalReport(int $totalCount, int $updatedResourcesCount): Report
+    {
+        $type = Report::TYPE_SUCCESS;
+        $template = 'CSV import successful: imported %d/%d row(s).';
+        $templateData = [
+            $updatedResourcesCount,
+            $totalCount,
+        ];
+
+        if (!empty($this->reports)) {
+            $type = $updatedResourcesCount === $totalCount
+                ? Report::TYPE_ERROR
+                : Report::TYPE_WARNING;
+            $template = 'Imported %d/%d row(s)';
+
+            [$warningsCount, $errorsCount] = $this->countReportTypes();
+
+            if ($warningsCount) {
+                $template .= '. %d warning(s)';
+                $templateData[] = $warningsCount;
+            }
+
+            if ($errorsCount) {
+                $template .= ', %d error(s)';
+                $templateData[] = $errorsCount;
+            }
+
+            $template .= '.';
```

Comment:
nitpick: Minor issue, if there are no warnings but only errors, the first part of the message will be separated with a comma instead of a full stop.

```suggestion
            $msg = [];

            [$warningCount, $errorCount] = $this->countReportTypes();

            if ($warningCount > 0) {
                $msg[] = '%d warning(s)';
                $templateData[] = $warningCount;
            }

            if ($errorCount > 0) {
                $msg[] = '%d error(s)';
                $templateData[] = $errorCount;
            }

            $template = 'Imported %d/%d row(s).' . (!empty($msg) ? ' ' . implode(', ', $msg) .'.' : '');
```